src/allocator.f90

```diff
@@ -59,39 +61,34 @@ module m_allocator
 
 contains
 
-  function allocator_init(mesh, sz) result(allocator)
-    type(mesh_t), target, intent(inout) :: mesh
-    integer, intent(in) :: sz
+  function allocator_init(nx, ny, nz, sz) result(allocator)
+    integer, intent(in) :: nx, ny, nz, sz
```

Comment:
Would it be better to pass the grid dimensions as a triple? e.g.
```
allocator_init([nx, ny, nz], sz)
```
this would prevent mixing up where `sz` goes.